src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/caseworker/transfercase/TransferCaseStayedService.java

```diff
@@ -0,0 +1,150 @@
+package uk.gov.hmcts.cmc.claimstore.services.ccd.callbacks.caseworker.transfercase;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.json.JSONArray;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.cmc.ccd.domain.CCDCase;
+import uk.gov.hmcts.cmc.ccd.domain.CaseEvent;
+import uk.gov.hmcts.cmc.claimstore.config.CaseStayedIncrementConfiguration;
+import uk.gov.hmcts.cmc.claimstore.models.idam.User;
+import uk.gov.hmcts.cmc.claimstore.requests.idam.IdamApi;
+import uk.gov.hmcts.cmc.claimstore.services.UserService;
+import uk.gov.hmcts.cmc.claimstore.services.ccd.CoreCaseDataService;
+import uk.gov.hmcts.cmc.domain.models.ClaimState;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@AllArgsConstructor
+@Service
+@Slf4j
+public class TransferCaseStayedService {
+
+    private final CoreCaseDataService coreCaseDataService;
+    private final UserService userService;
+    private final IdamApi idamApi;
+    private final CaseStayedIncrementConfiguration caseStayedIncrementConfiguration;
+
+    public void findCasesForTransfer() {
+
+        User user = userService.authenticateAnonymousCaseWorker();
+        String authorisation = user.getAuthorisation();
+        String userId = idamApi.retrieveUserDetails(authorisation).getId();
+
+        Integer pageNumber = caseStayedIncrementConfiguration.getPageIncrement();
+
+        log.info("Comparing cases to update into ccd");
+
+        if (pageNumber < getNumberOfPages(authorisation, userId)) {
+            compareCases(authorisation, userId, pageNumber);
+            caseStayedIncrementConfiguration.setPageIncrement(pageNumber + 1);
```

Comment:
I am not sure why we need this in configuration.

Reply:
Checking ccd each page..

Reply:
As discussed, we should not just process one page at one scheduled time. We should process all the pages on a given schedule time. And try to filter out the data in our search criteria.